idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,371 @@
+            return b.mass_solids[t] == b.volume_solid[t] * sum(
+                    b.solids[t].get_material_density_terms("Sol", j)
+                    for j in b.config.solid_property_package.component_list)
+
+        # Sum of mass fractions at all time equals 1
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Sum of mass fractions at all time")
+        def sum_component_constraint(b, t):
+            if t == b.flowsheet().config.time.first():
+                return Constraint.Skip
+            else:
+                return 1 == sum(b.solids[t].mass_frac_comp[j]
+                                for j in b.solids[t].
+                                _params.component_list)
+
+        if self.config.energy_balance_type != EnergyBalanceType.none:
+            # Solid phase energy balance
+            # Accumulation equal to heat transfer
+            self.solids_energy_holdup = Var(
+                    self.flowsheet().time,
+                    initialize=1,
+                    doc="Solid phase energy holdup",
+                    units=units_meta_solid('energy'))
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy holdup constraints")
+            def solids_energy_holdup_constraints(b, t):
+                return b.solids_energy_holdup[t] == (
+                      self.volume_solid[t] *
+                      b.solids[t].get_energy_density_terms("Sol"))
+
+            self.solids_energy_accumulation = DerivativeVar(
+                        self.solids_energy_holdup,
+                        wrt=self.flowsheet().config.time,
+                        doc="Solids energy accumulation",
+                        units=units_meta_solid('energy') /
+                            units_meta_solid('time'))
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy accumulation constraints")
+            def solids_energy_accumulation_constraints(b, t):
+                return b.solids_energy_accumulation[t] == \
+                    - sum(b.reactions[t].reaction_rate[r] *
+                          b.volume_solid[t] *
+                          b.reactions[t].dh_rxn[r]
+                          for r in b.config.reaction_package.
+                          rate_reaction_idx)
+        if self.config.energy_balance_type == EnergyBalanceType.none:
+            # Fix solids temperature to initial value for isothermal conditions
+            @self.Constraint(
+                self.flowsheet().config.time,
+                doc="Isothermal solid phase constraint")
+            def isothermal_solid_phase(b, t):
+                if t == b.flowsheet().config.time.first():
+                    return Constraint.Skip
+                else:
+                    return (
+                        b.solids[t].temperature ==
+                        b.solids[0].temperature)
+
```

Comment:
I missed this previously but please can you update the documentation with this.

Reply:
Yes, I'll update the documentation.